src/test/java/org/icatproject/ids/IcatReaderTest.java

```diff
@@ -0,0 +1,145 @@
+package org.icatproject.ids;
+
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.when;
+
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Properties;
+
+import org.icatproject.EntityBaseBean;
+import org.icatproject.Grouping;
+import org.icatproject.ICAT;
+import org.icatproject.IcatException_Exception;
+import org.icatproject.PublicStep;
+import org.icatproject.Rule;
+import org.icatproject.User;
+import org.icatproject.UserGroup;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+
+public class IcatReaderTest {
+
+    @Mock
+    private PropertyHandler mockedPropertyHandler;
+
+    private String icatUrl;
+    private ICAT icatService;
+    private List<String> readerCreds;
+    private String rootSessionId;
+
+    private List<EntityBaseBean> createdIcatObjects = new ArrayList<>();
+    private PublicStep publicStep;
+
+    @Before
+    public void setup() throws FileNotFoundException, IOException, IcatException_Exception {
+        Properties testProps = new Properties();
+        testProps.load(new FileInputStream("src/test/resources/test.properties"));
+        readerCreds = Arrays.asList(testProps.getProperty("login.reader").trim().split("\\s+"));
+        String rootCredsString = testProps.getProperty("login.root");
+        // serverUrl must be defined in the maven ~/.m2/settings.xml file
+        icatUrl = ICATGetter.getCleanUrl(System.getProperty("serverUrl"));
+        icatService = ICATGetter.getService(icatUrl);
+        rootSessionId = TestUtils.login(icatService, rootCredsString);
+        setupPropertyHandler();
+        createReaderAccessToPublicSteps();
+    }
+
+    private void setupPropertyHandler() {
+        MockitoAnnotations.initMocks(this);
+        when(mockedPropertyHandler.getIcatService()).thenReturn(icatService);
+        when(mockedPropertyHandler.getReader()).thenReturn(readerCreds);
+    }
+
+    @Test
+    public void testIcatReaderCreation() throws IcatException_Exception {
+        IcatReader icatReader = new IcatReader(mockedPropertyHandler);
+        assertNotNull("An ICAT session ID was expected", icatReader.getSessionId());
```

Comment:
This requires the user `reader` to be added in the authentication plugin. Until now, the documentation on how to run the tests refer to the corresponding documentation for `icat.server` which in turn instructs to add the users `notroot`, `piOne`, `piTwo`, `root`, `guest`, and `CIC`.